src/ImageSharp/Formats/Jpg/JpegDecoderCore.cs

```diff
@@ -18,237 +17,154 @@ namespace ImageSharp.Formats
-
-        /// <summary>
-        /// Saved state between progressive-mode scans.
-        /// </summary>
-        private readonly Block8x8F[][] progCoeffs;
-
-        /// <summary>
-        /// The huffman trees
-        /// </summary>
-        private readonly HuffmanTree[] huffmanTrees;
-
-        /// <summary>
-        /// Quantization tables, in zigzag order.
-        /// </summary>
-        private readonly Block8x8F[] quantizationTables;
-
-        /// <summary>
-        /// A temporary buffer for holding pixels
-        /// </summary>
-        private readonly byte[] temp;
-
-        /// <summary>
-        /// The byte buffer.
+        /// The App14 marker color-space
         /// </summary>
-        private Bytes bytes;
+        private byte adobeTransform;
 
         /// <summary>
-        /// The byte buffer.
+        /// Whether the image is in CMYK format with an App14 marker
         /// </summary>
-        private Stream inputStream;
+        private bool adobeTransformValid;
 
         /// <summary>
-        /// Holds the unprocessed bits that have been taken from the byte-stream.
+        /// The black image to decode to.
         /// </summary>
-        private Bits bits;
+        private JpegPixelArea blackImage;
 
         /// <summary>
-        /// The image width
+        /// A grayscale image to decode to.
         /// </summary>
-        private int imageWidth;
+        private JpegPixelArea grayImage;
 
         /// <summary>
-        /// The image height
+        /// The horizontal resolution. Calculated if the image has a JFIF header.
         /// </summary>
-        private int imageHeight;
+        private short horizontalResolution;
 
         /// <summary>
-        /// The number of color components within the image.
+        /// Whether the image has a JFIF header
         /// </summary>
-        private int componentCount;
+        private bool isJfif;
 
         /// <summary>
-        /// A grayscale image to decode to.
+        /// The vertical resolution. Calculated if the image has a JFIF header.
         /// </summary>
-        private GrayImage grayImage;
+        private short verticalResolution;
 
         /// <summary>
         /// The full color image to decode to.
         /// </summary>
         private YCbCrImage ycbcrImage;
 
         /// <summary>
-        /// The array of keyline pixels in a CMYK image
+        /// Initializes a new instance of the <see cref="JpegDecoderCore" /> class.
         /// </summary>
-        private byte[] blackPixels;
+        public JpegDecoderCore()
+        {
+            this.HuffmanTrees = HuffmanTree.CreateHuffmanTrees();
+            this.QuantizationTables = new Block8x8F[MaxTq + 1];
+            this.Temp = new byte[2 * Block8x8F.ScalarCount];
+            this.ComponentArray = new Component[MaxComponents];
+            this.ProgCoeffs = new Block8x8F[MaxComponents][];
+            this.Bits = default(Bits);
+            this.Bytes = Bytes.Create();
```

Comment:
I like explicit things better, when possible. Lazy loads are for the case, where you don't want to think about lifecycle, letting it "just happen" :smile: